Add main script

antoniotorresm opened the pull request:
Add initial script with two tests implemented:

* EnrollmentTest: try to enroll a number of clients simultaneously
* APITest: run a IPA CLI command a number of times simultaneously

This requires the host to have vagrant, libvirt, and ansible available
in order to run.

Signed-off-by: Antonio Torres <antorres@redhat.com>

antoniotorresm commented:
@rcritten @fcami

rcritten commented:
Can you add this as the script header:

#
# Copyright (C) 2021 FreeIPA Contributors see COPYING for license
#

I think this can be GPLv3 so add that COPYING file to the root as well.

rcritten commented on ipaperftest.py:
```diff
@@ -0,0 +1,485 @@
+        - "/var/log/httpd"
+        - "/var/log/dirsrv"
+        - "/var/log/krb5kdc.log"
+        - "/var/log/pki/pki-tomcat/ca/transactions"
```
perhaps the debug log instead of transactions

antoniotorresm replied:
Updated to gather the entire /var/log/pki/pki-tomcat/ca directory.

rcritten commented on ipaperftest.py:
```diff
@@ -0,0 +1,485 @@
+"""
+
+
+class EnrollmentTest:
```
Including the tests in one file is fine for starters IMHO. We can break them out into discoverable plugins later.

rcritten commented:
Can you also update the README with information on how to execute the tests?

antoniotorresm commented:
Updated PR:

* Added README
* Added COPYING
* Added gitignore
* Added IP address generator for private network
* Added support for passing private keys (some Vagrant images need a certain key to log in)
* Increased size of management network (so that we can have more than 255 clients)

rcritten commented on README.md:
```diff
@@ -1 +1,31 @@
+  * `APITest`: Try to run IPA command n times simultaneously.
+* `amount`: Size of the test. For example, in the case of `EnrollmentTest`, it will create this amount of clients. 
+* `command`: Command to run. Only relevant when using `APITest`. Use `{id}` for getting an unique ID into the command. Example: `ipa user-add user{id}`.
+* `client-image`: Vagrant image to use for the clients. Default: `antorres/fedora-34-ipa-client`. If you use a different image, make sure it has all the needed packages installed.
```
Should we define the minimum required packages? I guess it may depend on the test, but I suppose it would be `[free]ipa-client `and `at`.

antoniotorresm replied:
Thanks, added. Also added `rsync` to the list of packages needed in the controller (it is used to collect logs).

rcritten commented:
ack, looks good to me.
